pytorch_to_returnn/torch/_C.py

```diff
@@ -149,13 +150,19 @@ def originating_tensor_axis(self) -> int:
     return tensor_entry.torch_axis_from_returnn_axis[returnn_axis]
 
   def as_tensor(self):
+    if self.originating_tensor is None and self.merged_dims:
+      return numpy.prod([d.as_tensor() if d.dim_tag.dimension is None else int(d) for d in self.merged_dims])
     assert self.originating_tensor is not None
     from .nn.modules import Length
     tensor = Length(axis=self.originating_tensor_axis).as_returnn_torch_functional()(self.originating_tensor)
-    tensor.fill_(int(self))
-    tensor.is_defined = True
     naming = Naming.get_instance()
     tensor_entry = naming.tensors[tensor]
+    if tensor_entry.returnn_data.have_batch_axis():
```

Comment:
This is too specific. You want to apply this code always when it is not a scalar.
```suggestion
    if tensor_entry.returnn_data.batch_ndim > 0:
```

Reply:
Also, you don't need the `tensor_entry` here. Simply:
```suggestion
    if len(tensor.shape) > 0:
```